apollo-router/src/apollo_router.rs

```diff
@@ -1,143 +1,155 @@
+//         let plan_cache_limit = std::env::var("ROUTER_PLAN_CACHE_LIMIT")
+//             .ok()
+//             .and_then(|x| x.parse().ok())
+//             .unwrap_or(100);
+//         let query_cache_limit = std::env::var("ROUTER_QUERY_CACHE_LIMIT")
+//             .ok()
+//             .and_then(|x| x.parse().ok())
+//             .unwrap_or(100);
+//         let query_planner = Arc::new(CachingQueryPlanner::new(
+//             RouterBridgeQueryPlanner::new(Arc::clone(&schema)),
+//             plan_cache_limit,
+//         ));
+//
+//         // NaiveIntrospection instantiation can potentially block for some time
+//         let naive_introspection = {
+//             let schema = Arc::clone(&schema);
+//             tokio::task::spawn_blocking(move || NaiveIntrospection::from_schema(&schema))
+//                 .await
+//                 .expect("NaiveIntrospection instantiation panicked")
+//         };
+//
+//         // Start warming up the cache
+//         //
+//         // We don't need to do this in background because the old server will keep running until
+//         // this one is ready.
+//         //
+//         // If we first warm up the cache in foreground, then switch to the new config, the next
+//         // queries will benefit from the warmed up cache. While if we switch and warm up in
+//         // background, the next queries might be blocked until the cache is primed, so there'll be
+//         // a perf hit.
+//         if let Some(previous_router) = previous_router {
+//             for (query, operation, options) in previous_router
+//                 .query_planner_service
+//                 .get_ref()
+//                 .get_hot_keys()
+//                 .await
+//             {
+//                 // We can ignore errors because some of the queries that were previously in the
+//                 // cache might not work with the new schema
+//                 let _ = query_planner.get(query, operation, options).await;
+//             }
+//         }
+//
+//         Self {
+//             naive_introspection,
+//             query_planner_service: query_planner,
+//             service_registry,
+//             query_cache: Arc::new(QueryCache::new(query_cache_limit)),
+//             schema,
+//         }
+//     }
+// }
+//
+// #[async_trait::async_trait]
+// impl Router for ApolloRouter {
+//     type PreparedQuery = ApolloPreparedQuery;
+//
+//     #[tracing::instrument(level = "debug", skip_all)]
+//     async fn prepare_query(&self, context: Context) -> Result<ApolloPreparedQuery, Response> {
+//         if let Some(response) = self.naive_introspection.get(&context.request.body().query) {
+//             return Err(response);
+//         }
+//
+//         let query = self
+//             .query_cache
+//             .get_query(&context.request.body().query)
+//             .instrument(tracing::info_span!("query_parsing"))
+//             .await;
+//
+//         if let Some(query) = query.as_ref() {
```

Comment:
Make sure that validation is being performed elsewhere. I think it is missing at the moment.
This goes for validation of variable and query plan validation.

Reply:
isn't a test failing ? we can probably add one